packages/suite/src/hooks/suite/useTheme.ts

```diff
@@ -1,22 +1,32 @@
-import * as suiteActions from '@suite-actions/suiteActions';
-import { useActions, useSelector } from '@suite-hooks';
+import { useState, useEffect } from 'react';
+import { useSelector } from '@suite-hooks';
+import { getOsTheme } from '@suite-utils/env';
 import { getThemeColors } from '@suite-utils/theme';
-import { SuiteThemeColors } from '@suite-types';
+import { loadSuiteSettings } from '@suite-actions/storageActions';
+import type { SuiteState } from '@suite-reducers/suiteReducer';
 
 export const useTheme = () => {
-    const themeObj = useSelector(state => state.suite.settings.theme);
-    const { setTheme } = useActions({
-        setTheme: suiteActions.setTheme,
-    });
+    const [preloadedTheme, setPreloadedTheme] = useState<SuiteState['settings']['theme']>();
+    const { theme, storageLoaded } = useSelector(state => ({
+        theme: state.suite.settings.theme,
+        storageLoaded: state.suite.storageLoaded,
+    }));
 
-    const theme = getThemeColors(themeObj);
-    const themeVariant = themeObj.variant;
+    useEffect(() => {
+        if (storageLoaded || preloadedTheme) return;
+        loadSuiteSettings()
```

Comment:
just quick thought
`useTheme` hook is used in multiple components at once.
if conditions are not meet (not loaded, not preloaded) each instance of `useTheme` will call `loadSuiteSettings` > `setPreloadedTheme` no?

Reply:
actually i noticed that we are using 2 different hooks called `useTheme`
`import { useTheme } from '@suite-hooks';` - (this one) getting values from redux, used in ~20 components
and 
`import { useTheme } from '@suite/components';` - getting values from the context

Reply:
i think we should only use the context one (from '@suite/components') what do you think?

Reply:
and maybe rename this one to `useThemeContext` or something, since it should be used only to provide data to the context

Reply:
I agree with renaming to `useThemeContext`, it should be absolutely called at one place only. But using `useTheme` from `@trezor/components` is kind of extensive change, since these two hooks have different return types. So we have to change either all occurences in `@trezor/components` or in suite. I'll think it through.

Reply:
Result in https://github.com/trezor/trezor-suite/pull/4314/commits/36ce8198de72cfd03e6558ac7775a477d3b2e2ea. There is a lot of changed files, but functionally it should be equivalent.